modules/translator.py

```diff
@@ -216,7 +251,18 @@ def get_api_key(self, translator_key: str):
             }
             api_key = api_key_map.get(translator_key, "")
 
-        if not api_key and translator_key!= 'Google Translate':
+        if not api_key and translator_key not in ['Google Translate', 'Local OpenAI Server']:
             raise ValueError(f"API key not found for translator: {translator_key}")
 
-        return api_key
\ No newline at end of file
+        return api_key
+    
+    def get_base_url(self, translator_key: str):
+        base_url = None
+
+        if 'Local OpenAI' in translator_key:
+            base_url = self.settings.ui.llm_widgets['local_oai_url_input'].text()
+
+        if not base_url and translator_key == "Local OpenAI Server":
+            raise ValueError(f"Base URL not found for translator: {translator_key}")
+
+        return base_url
```

Comment:
Added logic to pull the `base_url` in `translator.py` and pass it to the `get_llm_client()` helper, rather than have the helper need to access the settings pane.